include/seqan3/range/view/single_pass_input.hpp

```diff
@@ -83,16 +81,33 @@ class single_pass_input_view : public detail::view_base
      * \{
      * \brief All standard functions are explicitly defaulted.
      */
+    //!\brief Default default-constructor.
     constexpr single_pass_input_view() = default;
+    //!\brief Default copy-constructor.
     constexpr single_pass_input_view(single_pass_input_view const &) = default;
+    //!\brief Default move-constructor.
     constexpr single_pass_input_view(single_pass_input_view &&) = default;
+    //!\brief Default copy-assignment.
     constexpr single_pass_input_view & operator=(single_pass_input_view const &) = default;
+    //!\brief Default move-assignment
     constexpr single_pass_input_view & operator=(single_pass_input_view &&) = default;
+    //!\brief Default destructor.
     ~single_pass_input_view() = default;
 
     //!\brief Construction from the underlying view.
-    single_pass_input_view(urng_t && urng) :
-        view_state_ptr{new view_state{std::forward<urng_t>(urng)}}
+    explicit single_pass_input_view(urng_t _urng) :
+        state_ptr{new state{std::move(_urng)}}
+    {}
+
+    //!\brief Construction from std::ranges::ViewableRange.
+    template <typename other_urng_t>
+    //!\cond
+    requires !std::Same<remove_cvref_t<other_urng_t>, single_pass_input_view> &&
+             std::ranges::ViewableRange<other_urng_t> &&  // Must come after self type check to avoid conflicts with the move constructor.
+             std::Constructible<urng_t, ranges::ref_view<std::remove_reference_t<other_urng_t>>>
+    //!\endcond
+    explicit single_pass_input_view(other_urng_t && _urng) :
+        single_pass_input_view{std::view::all(_urng)}
     {}
```

Comment:
How do you set the state of a default-constructed single_pass view with these constructors?

Reply:
As discussed. The pointer is reset with the new allocated state.